sklearn/notebooks/ml_engine_training.ipynb

```diff
@@ -0,0 +1,489 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 1: Create your python model file\n",
+    "\n",
+    "First, we'll create the python model file (provided below) that we'll upload to ML Engine. This is similar to your normal process for creating a scikit-learn model. However, there are two key differences:\n",
+    "1. Downloading the data from GCS at the start of your file, so that ML Engine can access the data.\n",
+    "1. Exporting/saving the model to GCS at the end of your file, so that you can use it for predictions.\n",
+    "\n",
+    "The code in this file loads the data into a pandas DataFrame that can be used by scikit-learn. Then the model is fit against the training data. Lastly, sklearn's built in version of joblib is used to save the model to a file that can be uploaded to [ML Engine's prediction service](https://cloud.google.com/ml-engine/docs/scikit/getting-predictions#deploy_models_and_versions).\n",
+    "\n",
+    "**REPLACE Line 18: BUCKET_ID = 'true-ability-192918' with your GCS BUCKET_ID**\n",
+    "\n",
+    "Note: In normal practice you would want to test your model locally on a small dataset to ensure that it works, before using it with your larger dataset on ML Engine. This avoids wasted time and costs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Writing ./census_training/train.py\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%writefile ./census_training/train.py\n",
+    "# [START setup]\n",
+    "import datetime\n",
+    "import os\n",
+    "import pandas as pd\n",
+    "import subprocess\n",
+    "\n",
+    "from google.cloud import storage\n",
+    "\n",
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "from sklearn.externals import joblib\n",
+    "from sklearn.feature_selection import SelectKBest\n",
+    "from sklearn.pipeline import FeatureUnion\n",
+    "from sklearn.pipeline import Pipeline\n",
+    "from sklearn.preprocessing import LabelBinarizer\n",
+    "\n",
+    "\n",
+    "# TODO: REPLACE 'true-ability-192918' with your GCS BUCKET_ID\n",
+    "BUCKET_ID = 'true-ability-192918'\n",
+    "# [END setup]\n",
+    "\n",
+    "\n",
+    "# ---------------------------------------\n",
+    "# 1. Add code to download the data from GCS (in this case, using the publicly hosted data).\n",
+    "# ML Engine will then be able to use the data when training your model.\n",
+    "# ---------------------------------------\n",
+    "# [START download-data]\n",
+    "census_data_filename = 'adult.data'\n",
+    "\n",
+    "# Public bucket holding the census data\n",
+    "bucket = storage.Client().bucket('cloud-samples-data')\n",
+    "\n",
+    "# Path to the data inside the public bucket\n",
+    "data_dir = 'ml-engine/sklearn/census_data/'\n",
```

Comment:
switch to gs://cloud-samples-data/ml-engine/census/data/adult.data.csv ?

Reply:
Done.